components/ledger/pkg/ledger/revert/reverter.go

```diff
@@ -0,0 +1,101 @@
+package revert
+
+import (
+	"context"
+	"sync"
+
+	"github.com/formancehq/ledger/pkg/core"
+	"github.com/formancehq/ledger/pkg/ledger/runner"
+	"github.com/formancehq/ledger/pkg/storage"
+	"github.com/formancehq/stack/libs/go-libs/errorsutil"
+	"github.com/formancehq/stack/libs/go-libs/metadata"
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrAlreadyReverted = errors.New("transaction already reverted")
+	ErrRevertOccurring = errors.New("revert already occurring")
+)
+
+type Store interface {
+	GetTransaction(ctx context.Context, id string) (*core.ExpandedTransaction, error)
+}
+
+type Runner interface {
+	Execute(
+		ctx context.Context,
+		script core.RunScript,
+		dryRun bool,
+		logComputer runner.LogComputer,
+	) (*core.ExpandedTransaction, *core.LogHolder, error)
+}
+
+type LogIngester interface {
+	QueueLog(log *core.LogHolder)
+}
+
+type Reverter struct {
+	sync.Map
+	store       Store
+	runner      Runner
+	logIngester LogIngester
+}
+
+func (r *Reverter) RevertTransaction(ctx context.Context, id string) (*core.ExpandedTransaction, error) {
+	_, loaded := r.Map.LoadOrStore(id, struct{}{})
```

Comment:
We really have to be sure that there is one ledger instance !

Reply:
Sorry, I don't understand ^^